internal/service/appstream/find.go

```diff
@@ -92,3 +92,25 @@ func FindImageBuilderByName(ctx context.Context, conn *appstream.AppStream, name
 
 	return result, nil
 }
+
+// FindUserByUserNameAndAuthType Retrieve a appstream fleet by Username and authentication type
+func FindUserByUserNameAndAuthType(ctx context.Context, conn *appstream.AppStream, username, authType string) (*appstream.User, error) {
+	input := &appstream.DescribeUsersInput{
+		AuthenticationType: aws.String(authType),
+	}
+
+	var user *appstream.User
+	resp, err := conn.DescribeUsersWithContext(ctx, input)
```

Comment:
When practitioners have a large number of AppStream users, there could be more than a single call to `DescribeUsersWithContext()` will return. We prefer using paginated versions of these functions, or in cases where the AWS API doesn't provide them, they can be generated. (See https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#resource-contribution-guidelines for guidelines) The generation code is documented at https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/generate/listpages/README.md